core/src/test/java/io/honeycomb/opentelemetry/android/HoneycombDeterministicSamplerTest.kt

```diff
@@ -0,0 +1,44 @@
+package io.honeycomb.opentelemetry.android
+
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.trace.SpanKind
+import io.opentelemetry.api.trace.TraceId
+import io.opentelemetry.context.Context
+import io.opentelemetry.sdk.trace.data.LinkData
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+
+@RunWith(Parameterized::class)
+class HoneycombDeterministicSamplerTest(private val args: TestArguments) {
+
+    companion object {
+        @JvmStatic
+        @Parameters
+        fun data(): List<TestArguments> {
+            return listOf(
+                TestArguments(0, SamplingDecision.DROP),
+                TestArguments(1, SamplingDecision.RECORD_AND_SAMPLE),
+                TestArguments(10, SamplingDecision.RECORD_AND_SAMPLE),
+                TestArguments(100, SamplingDecision.RECORD_AND_SAMPLE),
+            )
+        }
+    }
+
+    @Test
+    fun testSampler() {
+        // static trace id to ensure the inner traceIdRatio sampler always samples.
+        val traceId = TraceId.fromLongs(10L, 10L)
+        val context = Context.root()
+
+        val sampler = HoneycombDeterministicSampler(args.rate)
+        val result = sampler.shouldSample(context, traceId, "test", SpanKind.CLIENT, Attributes.empty(), emptyList<LinkData>().toMutableList())
```

Comment:
UNLIKE the Swift PR, here we JUST need a `Context` instead of a `SpanContext`. This seems to be a difference in the OTel APIs: the Swift one has a signature:

```swift
func shouldSample(
    parentContext: SpanContext?,
    traceId: TraceId,
    name: String,
    kind: SpanKind,
    attributes: [String: AttributeValue],
    parentLinks: [SpanData.Link]
) -> any Decision
```

whereas the Android one has the signature 

```kotlin
fun shouldSample(
    context: Context,
    traceId: String,
    name: String,
    spanKind: SpanKind,
    attributes: Attributes,
    parentLinks: MutableList<LinkData>
): SamplingResult
```

and there doesn't seem to be a relationship between `Context` and `SpanContext` in the Android SDK.